Sub/Src/ComputerVision/Nodes/RecvNode.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import cv2
+import socket
+import sys
+import io
+import csv
+import time
+
+from MechOS.message_passing.Nodes.node_base import node_base
+from datetime import datetime
+
+
+class RecvNode(node_base):
+    def __init__(self, MEM, IP):
+        
+        node_base.__init__(self, MEM, IP)
+
+        # Data Size Per Packet
+        self.MAX_UDP_PACKET_SIZE = 1500
+        
+        # Resize image
+        self.resize_image = True
+        
+        # Collect Images (Switch to False for speed)
+        self.save_image = False
+   
+        self.ramBuffer = b''
+
+    def run(self):
+
+        while(True):
+            # Socket Max receive
+            # packet = sock.recv(MAX_UDP_PACKET_SIZE)
+            packet = self._recv('CAMERA', local=False)
+            #print(packet)
+
+            if packet:
+                # Buffer object (concatenated byteString)
+                self.ramBuffer += packet
+        
+                # Triple CRC bytestring and EOF byte
+                if (self.ramBuffer.endswith(b'\xc0\xc0\xc0\xc0')):
+        
+                    # Decapsulate Byte String (4 sets of 2 chars)
+                    self.ramBuffer = self.ramBuffer[:-4]
+        
+                    # Image integrity Data Check (imperfect)
+                    if ( self.ramBuffer.startswith(b'\xff\xd8') and self.ramBuffer.endswith(b'\xff\xd9') ):
+        
+                        # Capture Bytes
+#                        print('Recieved Bytes', ramBuffer)
+#                        input('press enter to continue...')
+        
+                        img_frame = cv2.imdecode(np.frombuffer(self.ramBuffer, dtype=np.uint8), 1)
+        
+                        # Our operations on the frame are here (if any)
+        
+                        if self.save_image:
```

Comment:
`save_image` looks like it isn't set anywhere else other than the init. I think it would be better to have this as something that is configurable during runtime or have it so that it can be toggled with a button push

Reply:
@PierceATronics Parameter Server Integration